doc/dev/CustomRepoGuide.md

```diff
@@ -0,0 +1,387 @@
+
+First thing you have to to, is to register on https://travis-ci.org . If it is not part of the registration process, you now have to link your github and your travis account. Google it, if you get stuck here, but as far as I remember, it was not hard to accomplish. If this is done, open your profile on travis-ci.org . You should now see all your repositories listed and marked with a gray "x". Just click the gray field to activate travis for your repository. All you need to do now, is to provide a file named `.travis.yml` (don't miss the dot at the front) to your projects root repository, which configures your test environment.
+
+I will show you one possible configuration first, before we go through each command step by step:
+
+~~~yml
+language: cpp
+sudo: false
+services: docker
+dist: trusty
+
+matrix:
+    include:
+        - compiler: clang
+          env: BUILD_TYPE=Release
+        - compiler: g++
+          env: BUILD_TYPE=Release
+        - compiler: g++
+          env: BUILD_TYPE=Debug COVERAGE=--coverage
+        - compiler: clang
+          env: BUILD_TYPE=Release WERROR=-Werror
+        - compiler: g++
+          env: BUILD_TYPE=Release WERROR=-Werror
+    allow_failures:
+        - env: BUILD_TYPE=Release WERROR=-Werror
+
+before_install:
+    - travis_retry timeout 120 docker pull nuto/nuto_docker:dev
+    - docker run -itd --name dock -u 0 -v $(pwd):/home/nuto/source nuto/nuto_docker:dev
+
+script:
+    - docker exec dock cmake -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_CXX_FLAGS="$COVERAGE $WERROR" ../source
+    - docker exec dock make -j2
+    - docker exec dock ctest
+~~~
+
+So the header is quiet self explanatory:
+
+~~~yml
+language: cpp
+sudo: false
+services: docker
+dist: trusty
+~~~
+
+We want to test something written in cpp, do not need sudo permissions on the server and request the additional docker (virtual machine) service on a ubuntu trusty (14.04) system.
+
+The next part sets your different test configurations
+
+~~~yml
+matrix:
+    include:
+        - compiler: clang
+          env: BUILD_TYPE=Release
+        - compiler: g++
+          env: BUILD_TYPE=Release
+        - compiler: g++
+          env: BUILD_TYPE=Debug COVERAGE=--coverage
+        - compiler: clang
+          env: BUILD_TYPE=Release WERROR=-Werror
+        - compiler: g++
+          env: BUILD_TYPE=Release WERROR=-Werror
+    allow_failures:
+        - env: BUILD_TYPE=Release WERROR=-Werror
+~~~
+
+The `include:` section covers the different build types you want to use. The 'compiler:' flag is obvious. The `env:` flag sets environmental variables that you use later inside instructions and if statements. The `allow_failures:` section specifies build setups, that are allowed to fail. So the rating if your test passed or not does not depend on this setup. In the example above, we want two builds to fail, if the compiler creates any warnings, but we don't want those builds to cause a failure of the whole test series. 
```

Comment:
"The `allow_failures:` section specifies jobs that are allowed to fail, i.e. the rating whether your build passed does not depend on these jobs. In the example above, we want two jobs to fail if the compiler creates any warnings, but we don't want those jobs to cause a failure of the whole build."